File: src/types/event.ts
// Event types - aligned with ARCHITECTURE.md database schema
export interface CalendarEvent {
  id: number;
  title: string;
  description?: string | null;
  start_time: string;
  end_time: string;
  reminder_time?: string | null;
  recurrence?: string | null;
  created_at: string;
}

export interface CreateEvent {
  title: string;
  description?: string | null;
  start_time: string;
  end_time: string;
  reminder_time?: string | null;
  recurrence?: string | null;
}

export interface UpdateEvent {
  title?: string;
  description?: string | null;
  start_time?: string;
  end_time?: string;
  reminder_time?: string | null;
  recurrence?: string | null;
}

File: src-tauri/src/models/event.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CalendarEvent {
    pub id: i64,
    pub title: String,
    pub description: Option<String>,
    pub start_time: String,
    pub end_time: String,
    pub reminder_time: Option<String>,
    pub recurrence: Option<String>,
    pub created_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateEvent {
    pub title: String,
    pub description: Option<String>,
    pub start_time: String,
    pub end_time: String,
    pub reminder_time: Option<String>,
    pub recurrence: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateEvent {
    pub title: Option<String>,
    pub description: Option<String>,
    pub start_time: Option<String>,
    pub end_time: Option<String>,
    pub reminder_time: Option<String>,
    pub recurrence: Option<String>,
}

